Move DetectorConstruction and ParallelWorld msgs under the logger system

**Is your feature request related to a problem? Please describe.**

Now that I'm debugging with setting the logging to very high numbers, I still see the "Adding .. to parallel world" msgs, etc.

**Describe the solution you'd like**

I'd like to move 
https://github.com/LDMX-Software/ldmx-sw/blob/trunk/SimCore/src/SimCore/ParallelWorld.cxx#L27
and
https://github.com/LDMX-Software/ldmx-sw/blob/trunk/SimCore/src/SimCore/DetectorConstruction.cxx#L141
under the `ldmx_log(debug)`

Not as trivial as I thought:
 * adding just the debug doesnt work
 * then adding `enableLogging` in the respective headers I get a long msg about lambda functions not working, see details below:

<details>

```

In file included from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:22,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Conditions.h:21,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/EventProcessor.h:13,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/ConditionsInterface.h:12,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/Geo/Parser.h:11,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/DetectorConstruction.h:13,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:1:
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx: In lambda function:
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:146:37: error: 'this' was not captured for this lambda function
  146 | #define ldmx_log(lvl) BOOST_LOG_SEV(theLog_, ::framework::logging::level::lvl)
      |                                     ^~~~~~~
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:176:9: note: in expansion of macro 'ldmx_log'
  176 |         ldmx_log(debug)
      |         ^~~~~~~~
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:146:37: error: 'this' was not captured for this lambda function
  146 | #define ldmx_log(lvl) BOOST_LOG_SEV(theLog_, ::framework::logging::level::lvl)
      |                                     ^~~~~~~
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:176:9: note: in expansion of macro 'ldmx_log'
  176 |         ldmx_log(debug)
      |         ^~~~~~~~
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx: In instantiation of 'simcore::DetectorConstruction::ConstructSDandField()::<lambda(auto:2)> [with auto:2 = std::shared_ptr<simcore::XsecBiasingOperator>]':
/usr/include/c++/11/bits/stl_algo.h:3820:5:   required from '_Funct std::for_each(_IIter, _IIter, _Funct) [with _IIter = __gnu_cxx::__normal_iterator<const std::shared_ptr<simcore::XsecBiasingOperator>*, std::vector<std::shared_ptr<simcore::XsecBiasingOperator>, std::allocator<std::shared_ptr<simcore::XsecBiasingOperator> > > >; _Funct = simcore::DetectorConstruction::ConstructSDandField()::<lambda(auto:2)>]'
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/Factory.h:284:18:   required from 'void simcore::Factory<Prototype, PrototypePtr, PrototypeConstructorArgs>::apply(UnaryFunction) const [with UnaryFunction = simcore::DetectorConstruction::ConstructSDandField()::<lambda(auto:2)>; Prototype = simcore::XsecBiasingOperator; PrototypePtr = std::shared_ptr<simcore::XsecBiasingOperator>; PrototypeConstructorArgs = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const framework::config::Parameters&}]'
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:152:53:   required from here
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:146:37: error: 'this' was not captured for this lambda function
  146 | #define ldmx_log(lvl) BOOST_LOG_SEV(theLog_, ::framework::logging::level::lvl)
      |                                     ^
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:176:9: note: in expansion of macro 'ldmx_log'
  176 |         ldmx_log(debug)
      |         ^~~~~~~~
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:146:37: error: invalid use of non-static data member 'simcore::DetectorConstruction::theLog_'
  146 | #define ldmx_log(lvl) BOOST_LOG_SEV(theLog_, ::framework::logging::level::lvl)
      |                                     ^
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:176:9: note: in expansion of macro 'ldmx_log'
  176 |         ldmx_log(debug)
      |         ^~~~~~~~
In file included from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Conditions.h:21,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/EventProcessor.h:13,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/ConditionsInterface.h:12,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/Geo/Parser.h:11,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/DetectorConstruction.h:13,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:1:
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:137:40: note: declared here
  137 |   mutable ::framework::logging::logger theLog_{ \
      |                                        ^~~~~~~
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:137:40: note: in definition of macro 'enableLogging'
  137 |   mutable ::framework::logging::logger theLog_{ \
      |                                        ^~~~~~~
In file included from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:22,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Conditions.h:21,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/EventProcessor.h:13,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/ConditionsInterface.h:12,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/Geo/Parser.h:11,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/DetectorConstruction.h:13,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:1:
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:146:37: error: 'this' was not captured for this lambda function
  146 | #define ldmx_log(lvl) BOOST_LOG_SEV(theLog_, ::framework::logging::level::lvl)
      |                                     ^
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:176:9: note: in expansion of macro 'ldmx_log'
  176 |         ldmx_log(debug)
      |         ^~~~~~~~
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:146:37: error: invalid use of non-static data member 'simcore::DetectorConstruction::theLog_'
  146 | #define ldmx_log(lvl) BOOST_LOG_SEV(theLog_, ::framework::logging::level::lvl)
      |                                     ^
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:176:9: note: in expansion of macro 'ldmx_log'
  176 |         ldmx_log(debug)
      |         ^~~~~~~~
In file included from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Conditions.h:21,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/EventProcessor.h:13,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/ConditionsInterface.h:12,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/Geo/Parser.h:11,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/include/SimCore/DetectorConstruction.h:13,
                 from /sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/SimCore/src/SimCore/DetectorConstruction.cxx:1:
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:137:40: note: declared here
  137 |   mutable ::framework::logging::logger theLog_{ \
      |                                        ^~~~~~~
/sdf/home/t/tamasvami/BugFixes/temp/ldmx-sw/Framework/include/Framework/Logger.h:137:40: note: in definition of macro 'enableLogging'
  137 |   mutable ::framework::logging::logger theLog_{ \
      |                                        ^~~~~~~

 ```

</details>

--- a/DQM/src/DQM/TrigScintDQM.cxx
+++ b/DQM/src/DQM/TrigScintDQM.cxx
@@ -11,7 +11,7 @@ TrigScintDQM::TrigScintDQM(const std::string &name, framework::Process &process)
 TrigScintDQM::~TrigScintDQM() {}
 
 void TrigScintDQM::onProcessStart() {
-  std::cout << "Process starts!" << std::endl;
+  ldmx_log(debug) << "Process starts!";
 
   getHistoDirectory();
 
@@ -42,8 +42,8 @@ void TrigScintDQM::configure(framework::config::Parameters &ps) {
   hitCollectionName_ = ps.getParameter<std::string>("hit_collection");
   padName_ = ps.getParameter<std::string>("pad");
 
-  std::cout << "In TrigScintDQM::configure, got parameters "
-            << hitCollectionName_ << " and " << padName_ << std::endl;
+  ldmx_log(debug) << "In TrigScintDQM::configure, got parameters "
+                  << hitCollectionName_ << " and " << padName_;
 }
 
 void TrigScintDQM::analyze(const framework::Event &event) {

--- a/DQM/src/DQM/TrigScintHitDQM.cxx
+++ b/DQM/src/DQM/TrigScintHitDQM.cxx
@@ -9,7 +9,7 @@ TrigScintHitDQM::TrigScintHitDQM(const std::string &name,
 TrigScintHitDQM::~TrigScintHitDQM() {}
 
 void TrigScintHitDQM::onProcessStart() {
-  std::cout << "Process starts!" << std::endl;
+  ldmx_log(debug) << "Process starts!";
 
   getHistoDirectory();
 
@@ -49,8 +49,8 @@ void TrigScintHitDQM::configure(framework::config::Parameters &ps) {
   hitCollectionName_ = ps.getParameter<std::string>("hit_collection");
   padName_ = ps.getParameter<std::string>("pad").c_str();
 
-  std::cout << "In TrigScintHitDQM::configure, got parameters "
-            << hitCollectionName_ << " and " << padName_ << std::endl;
+  ldmx_log(debug) << "In TrigScintHitDQM::configure, got parameters "
+                  << hitCollectionName_ << " and " << padName_;
 }
 
 void TrigScintHitDQM::analyze(const framework::Event &event) {

--- a/SimCore/include/SimCore/DetectorConstruction.h
+++ b/SimCore/include/SimCore/DetectorConstruction.h
@@ -73,6 +73,9 @@ class DetectorConstruction : public G4VUserDetectorConstruction {
 
   /// interface to conditions to be passed to SDs
   simcore::ConditionsInterface &conditions_interface_;
+
+  /// Enable logging
+  enableLogging("DetectorConstruction")
 };  // DetectorConstruction
 }  // namespace simcore
 

--- a/SimCore/include/SimCore/ParallelWorld.h
+++ b/SimCore/include/SimCore/ParallelWorld.h
@@ -45,6 +45,9 @@ class ParallelWorld : public G4VUserParallelWorld {
   /** The auxiliary GDML info reader. */
   geo::AuxInfoReader* auxInfoReader_{nullptr};
 
+  /// Enable logging
+  enableLogging("ParallelWorld")
+
 };  // ParallelWorld
 }  // namespace simcore
 

--- a/SimCore/src/SimCore/DetectorConstruction.cxx
+++ b/SimCore/src/SimCore/DetectorConstruction.cxx
@@ -138,9 +138,8 @@ void DetectorConstruction::ConstructSDandField() {
     // attach to volumes
     for (G4LogicalVolume* volume : *G4LogicalVolumeStore::GetInstance()) {
       if (sd->isSensDet(volume)) {
-        std::cout << "[ DetectorConstruction ] : "
-                  << "Attaching " << sd->GetName() << " to "
-                  << volume->GetName() << std::endl;
+        ldmx_log(debug) << "Attaching " << sd->GetName() << " to "
+                        << volume->GetName();
         volume->SetSensitiveDetector(sd);
       }
     }
@@ -149,7 +148,7 @@ void DetectorConstruction::ConstructSDandField() {
   // Biasing operators were created in RunManager::setupPhysics
   //  which is called before G4RunManager::Initialize
   //  which is where this method ends up being called.
-  simcore::XsecBiasingOperator::Factory::get().apply([](auto bop) {
+  simcore::XsecBiasingOperator::Factory::get().apply([&](auto bop) {
     logical_volume_tests::Test includeVolumeTest{nullptr};
     if (bop->getVolumeToBias().compare("ecal") == 0) {
       includeVolumeTest = &logical_volume_tests::isInEcal;
@@ -173,9 +172,8 @@ void DetectorConstruction::ConstructSDandField() {
       auto volume_name = volume->GetName();
       if (includeVolumeTest(volume, bop->getVolumeToBias())) {
         bop->AttachTo(volume);
-        std::cout << "[ DetectorConstruction ]: "
-                  << "Attaching biasing operator " << bop->GetName()
-                  << " to volume " << volume->GetName() << std::endl;
+        ldmx_log(debug) << "Attaching biasing operator " << bop->GetName()
+                        << " to volume " << volume->GetName();
       }  // BOP attached to target or ecal
     }    // loop over volumes
   });    // loop over biasing operators

--- a/SimCore/src/SimCore/ParallelWorld.cxx
+++ b/SimCore/src/SimCore/ParallelWorld.cxx
@@ -24,8 +24,8 @@ void ParallelWorld::ConstructSD() {
 
   for (int index = 0; index < parallelWorldLogical->GetNoDaughters(); index++) {
     G4VPhysicalVolume *physicalVol = parallelWorldLogical->GetDaughter(index);
-    std::cout << "[ ParallelWorld ]: Adding : " << physicalVol->GetName()
-              << " to parallel world." << std::endl;
+    ldmx_log(debug) << "Adding : " << physicalVol->GetName()
+                    << " to parallel world.";
     worldLogical->AddDaughter(physicalVol);
   }
 

